lib/ruby_lsp/requests/semantic_highlighting.rb

```diff
@@ -305,6 +305,8 @@ def visit_binary(node)
         return unless node.operator == :=~
 
         left = node.left
+        return unless left.is_a?(SyntaxTree::RegexpLiteral)
```

Comment:
Can we add a comment here similar to the one in the previous early return?
```suggestion
        # The regexp needs to be on the left hand side of the =~ for local variable capture
        return unless left.is_a?(SyntaxTree::RegexpLiteral)
```